autoload/editorconfig_core/ini.vim

```diff
@@ -60,23 +60,24 @@ lockvar s:SECTCRE s:OPTCRE s:MAX_SECTION_NAME s:MAX_PROPERTY_NAME s:MAX_PROPERTY
 " Read \p config_filename and return the options applicable to
 " \p target_filename.  This is the main entry point in this file.
 function! editorconfig_core#ini#read_ini_file(config_filename, target_filename)
-    let l:oldenc = &encoding
-
     if !filereadable(a:config_filename)
         return {}
     endif
 
-    try     " so &encoding will always be reset
-        let &encoding = 'utf-8'     " so readfile() will strip BOM
+    try
         let l:lines = readfile(a:config_filename)
+        if &encoding !=? 'utf-8'
+            " strip BOM
+            if len(l:lines) > 0 && l:lines[0][:2] ==# "\xEF\xBB\xBF"
```

Comment:
Will the first three bytes always come through as the UTF-8 BOM values?  E.g., what happens when the file is UTF-16?

Reply:
This part strips the UTF-8 BOM only when it exists. Otherwise it is kept as is.